File: app/src/main/java/com/alamodrafthouse/injection/module/AppModule.java
package com.alamodrafthouse.injection.module;

import com.alamodrafthouse.AlamoApplication;
import dagger.Module;
import dagger.Provides;

/**
 * Created by Shekar on 3/3/17.
 */

@Module
public class AppModule {
    AlamoApplication application;

    public AppModule(AlamoApplication app) {
        application = app;
    }

    @Provides AlamoApplication provideBaseApplication() {
        return application;
    }
}

File: app/src/main/java/com/mutualmobile/praxis/injection/module/AppModule.kt
package com.mutualmobile.praxis.injection.module

import com.mutualmobile.praxis.utils.CoroutineDispatcherProvider
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import kotlinx.coroutines.CoroutineDispatcher
import kotlinx.coroutines.Dispatchers
import javax.inject.Singleton

/**
 * Created by Rooparsh Kalia on 30 Jul 2021
 */

@Module
@InstallIn(SingletonComponent::class)
class AppModule {

    @Provides
    @Singleton
    internal fun provideSchedulers(): CoroutineDispatcherProvider {
        return object : CoroutineDispatcherProvider {
            override fun io(): CoroutineDispatcher = Dispatchers.IO
            override fun main(): CoroutineDispatcher = Dispatchers.Main
        }
    }
}
